library/systemd/src/lib/yast2/systemd/service.rb

```diff
@@ -0,0 +1,216 @@
+    #
+    #    # or using the full unit id 'sshd.service'
+    #
+    #    service = Yast::Systemd::Service.find('sshd.service')
+    #
+    #    ## If you can't handle any nil at the place of calling,
+    #    ## use the finder with exclamation mark;
+    #    ## Systemd::ServiceNotFound exception will be raised
+    #
+    #    service = Yast::Systemd::Service.find!('IcanHasMoar') # Systemd::ServiceNotFound: Service unit 'IcanHasMoar' not found
+    #
+    #    ## Get basic unit properties
+    #
+    #    service.unit_name   # 'sshd'
+    #    service.unit_type   # 'service'
+    #    service.id          # 'sshd.service'
+    #    service.description # 'OpenSSH Daemon'
+    #    service.path        # '/usr/lib/systemd/system/sshd.service'
+    #    service.loaded?     # true if it's loaded, false otherwise
+    #    service.running?    # true if it's active and running
+    #    service.enabled?    # true if enabled, false otherwise
+    #    service.disabled?   # true if disabled, false otherwise
+    #    service.status      # the same text output you get with `systemctl status sshd.service`
+    #    service.show        # equivalent of calling `systemctl show sshd.service`
+    #
+    #    ## Service unit file commands
+    #
+    #    # Unit file commands do modifications on the service unit. Calling them triggers
+    #    # service properties reloading. If a command fails, the error message is available
+    #    # through the method #error as a string.
+    #
+    #    service.start       # true if unit has been activated successfully
+    #    service.stop        # true if unit has been deactivated successfully
+    #    service.enable      # true if unit has been enabled successfully
+    #    service.disable     # true if unit has been disabled successfully
+    #    service.error       # error string available if some of the actions above fails
+    #
+    #    ## Extended service properties
+    #
+    #    # In case you need more details about the service unit than the default ones,
+    #    # you can extend the parameters for .find method. Those properties are
+    #    # then available on the service unit object under the #properties instance method.
+    #    # An extended property is always a string, you must convert it manually,
+    #    # no automatical casting is done by yast.
+    #    # To get an overview of available service properties, try e.g., `systemctl show sshd.service`
+    #
+    #    service = Yast::Systemd::Service.find('sshd', :type=>'Type')
+    #    service.properties.type  # 'simple'
+    class Service < Unit
+      Yast.import "Stage"
+      include Yast::Logger
+
+      UNIT_SUFFIX = ".service".freeze
+
+      class << self
+        # @param service_name [String] "foo" or "foo.service"
+        # @param propmap [Systemd::Unit::PropMap]
+        # @return [Service,nil] `nil` if not found
+        def find(service_name, propmap = {})
```

Comment:
It looks strange that `propmap` is expected to be an instance of `Yast2::Systemd::Unit::Propmap` but the default value is a hash.

Reply:
I agree. Shall we change it now? WDYT?

Reply:
No, as a Propmap is basically a Hash, let's keep it out of the scope of this PR.

Reply:
Ups, sorry. I did not read that, and [it is already changed at the same time to extract related `Yast2::Systemd::Unit` classes to their own files](https://github.com/yast/yast-yast2/pull/799/commits/00c3fb703d2db27f25ec546f9b0ef1803571e13e).